integration-tests/tests/e2e.rs

```diff
@@ -442,3 +430,240 @@ testkey
 
     Ok(())
 }
+
+async fn test_per_device_serviceinfo<F>(
```

Comment:
rename this  one to  `test_e2e_impl_per_device_serviceinfo` or something like that so that we know that is a copy of the original  `test_e2e_impl` with the per-device changes